Tests/StreamChatTests/Query/Sorting/ListDatabaseObserver+Sorting.swift

```diff
@@ -0,0 +1,301 @@
+        let expectation = self.expectation(description: "Observer notifies")
+        observer.onDidChange = { changes in
+            XCTAssertEqual(changes.count, 5)
+            expectation.fulfill()
+        }
+
+        try createChannels(mapping: defaultChannels)
+
+        waitForExpectations(timeout: defaultTimeout)
+
+        XCTAssertEqual(observer.items.count, 5)
+        XCTAssertEqual(observer.items.map(\.name), ["D", "B", "E", "A", "C"])
+    }
+
+    func test_timeToProcessMultipleChatChannels_customSorting_foreground() throws {
+        measureMetrics([.wallClockTime], automaticallyStartMeasuring: false) {
+            do {
+                try test_timeToProcessMultipleChatChannels_customSorting(isBackground: false)
+            } catch {
+                XCTFail(error.localizedDescription)
+            }
+        }
+    }
+
+    func test_timeToProcessMultipleChatChannels_customSorting_background() throws {
+        measureMetrics([.wallClockTime], automaticallyStartMeasuring: false) {
+            do {
+                try test_timeToProcessMultipleChatChannels_customSorting(isBackground: true)
+            } catch {
+                XCTFail(error.localizedDescription)
+            }
+        }
+    }
+
+    func test_timeToProcessMultipleChatChannels_customSorting(isBackground: Bool) throws {
+        let expectation = self.expectation(description: "observer is notified")
+
+        let sorting: [Sorting<ChannelListSortingKey>] = [
+            .init(key: .createdAt, isAscending: false),
+            .init(key: .custom(keyPath: \.name, key: "name"), isAscending: false)
+        ]
+
+        createObserver(with: sorting, isBackground: isBackground)
+        try observer.startObserving()
+
+        observer.onDidChange = { changes in
+            expectation.fulfill()
+            XCTAssertEqual(changes.count, self.bulkChannels.count)
+            XCTAssertEqual(self.observer.items.count, self.bulkChannels.count)
+        }
+
+        startMeasuring()
+        try createChannels(mapping: bulkChannels)
+
+        wait(for: [expectation])
+        stopMeasuring()
+    }
+
+    func test_timeToProcessMultipleChatChannels_defaultSorting_foreground() throws {
+        measureMetrics([.wallClockTime], automaticallyStartMeasuring: false) {
+            do {
+                try test_timeToProcessMultipleChatChannels_defaultSorting(isBackground: false)
+            } catch {
+                XCTFail(error.localizedDescription)
+            }
+        }
+    }
+
+    func test_timeToProcessMultipleChatChannels_defaultSorting_background() throws {
+        measureMetrics([.wallClockTime], automaticallyStartMeasuring: false) {
+            do {
+                try test_timeToProcessMultipleChatChannels_defaultSorting(isBackground: true)
+            } catch {
+                XCTFail(error.localizedDescription)
+            }
+        }
+    }
+
+    func test_timeToProcessMultipleChatChannels_defaultSorting(isBackground: Bool) throws {
```

Comment:
nit: I think having those private and instead of the `test_` prefix use `assert` makes more sense as they are being reused. Passing `#file` and `#line` in their asserts will help more identifying which test case fails (when and if it does)